spec/unit/hg_resolver_spec.cr

```diff
@@ -0,0 +1,189 @@
+require "./spec_helper"
+
+private def resolver(name)
+  Shards::HgResolver.new(name, hg_url(name))
+end
+
+module Shards
+  # Allow overriding `source` for the specs
+  class HgResolver
+    def source=(@source)
+    end
+  end
+
+  describe HgResolver do
+    before_each do
+      create_hg_repository "empty"
+      create_hg_commit "empty", "initial release"
+
+      create_hg_repository "unreleased"
+      create_hg_version_commit "unreleased", "0.1.0"
+      checkout_new_hg_branch "unreleased", "branch"
+      create_hg_commit "unreleased", "testing"
+      checkout_hg_rev "unreleased", "default"
+
+      create_hg_repository "unreleased-bm"
+      create_hg_version_commit "unreleased-bm", "0.1.0"
+      checkout_new_hg_bookmark "unreleased-bm", "branch"
+      create_hg_commit "unreleased-bm", "testing"
+      checkout_hg_rev "unreleased-bm", "default"
+
+      create_hg_repository "library", "0.0.1", "0.1.0", "0.1.1", "0.1.2", "0.2.0"
+
+      # Create a version tag not prefixed by 'v' which should be ignored
+      create_hg_tag "library", "99.9.9"
+    end
+
+    it "normalizes github bitbucket gitlab sources" do
```

Comment:
I missed this: why is it testing the `GitResolver` here? It should test the `HgResolver` instead, whatever that means for Mercurial.

Reply:
Sorry, these specs have been introduced by the last merge and I missed them. Should be fixed in the latest commit.